src/services/VideoEmbed.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace viget\videoembed\services;
+
+use Craft;
+
+class VideoEmbed
+{
+    /**
+     * Take a youtube or vimeo url and return the embed url
+     *
+     * @param string $url
+     * @return string
+     */
+    public function getEmbedUrl(string $url): ?string
+    {
+        if ($this->_isYoutube($url)) {
+            $urlParts = parse_url($url);
+            $query = $urlParts['query'] ?? null;
+
+            if ($query === null) return null;
+
+            parse_str($query, $segments);
+            $v = $segments['v'] ?? null;
+
+            if ($v === null) return null;
+
+            return '//www.youtube.com/embed/' . $v;
+        } else if ($this->_isShortYoutube($url)) {
+            $urlParts = parse_url($url);
+            $path = $urlParts['path'] ?? null;
+
+            if ($path === null) return null;
+
+            return '//www.youtube.com/embed' . $path;
+        } else if ($this->_isVimeo($url)) {
+            $urlParts = parse_url($url);
+            $path = $urlParts['path'] ?? null;
+
+            if ($path === null) return null;
+
+            $segments = explode('/', $path);
+            $firstSegment = $segments[1] ?? null;
+
+            if ($firstSegment === null) return null;
+            
+            return '//player.vimeo.com/video/' . $firstSegment . '?player_id=video&api=1';
+        } else {
+            return null;
+        }
+    }
+
+    /**
+     * Determine whether the url is a youtube or vimeo url
+     * @param string $url
+     * @return boolean
+     */
+    public function isVideoUrl(string $url): bool
```

Comment:
Is this function going to be exposed to the consuming app? (And then we'd use it to determine whether or not to show an iframe.)

Reply:
Correct, the entire class is exposed to Twig at `craft.videoEmbed` so we can access this method